tutorials/dapt-curation/code/utils.py

```diff
@@ -0,0 +1,389 @@
+
+    Args:
+        dataset (DocumentDataset): The dataset to be filtered.
+
+    Returns:
+        DocumentDataset: The filtered dataset.
+    """
+    filters = Sequential(
+        [
+            # If a document contains a number of words not
+            # within a specified range then discard
+            ScoreFilter(
+                WordCountFilter(min_words=5, max_words=100000),
+                text_field="text",
+                score_field="word_count",
+                score_type=int,
+            ),
+        ]
+    )
+    filtered_dataset = filters(dataset)
+    return filtered_dataset
+
+
+def filter_code_dataset(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Filters the given code dataset based on various code-specific criteria.
+    Refer to the full list of all filters here:
+    https://github.com/NVIDIA/NeMo-Curator/blob/main/config/heuristic_filter_code.yaml
+    https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/filters/code.py
+
+    Args:
+        dataset (DocumentDataset): The dataset to be filtered.
+
+    Returns:
+        DocumentDataset: The filtered dataset.
+    """
+    filters = Sequential(
+        [
+            # if the comment to code ratio is not within the specified range,
+            # discard the document
+            ScoreFilter(
+                PythonCommentToCodeFilter(
+                    min_comment_to_code_ratio=0.001, max_comment_to_code_ratio=0.80
+                ),
+                text_field="text",
+                score_type=float,
+            ),
+            # if the number of lines of code is not within the specified range,
+            # discard the document
+            ScoreFilter(
+                NumberOfLinesOfCodeFilter(min_lines=5, max_lines=20000),
+                text_field="text",
+                score_type=int,
+            ),
+        ]
+    )
+    filtered_dataset = filters(dataset)
+    return filtered_dataset
+
+
+# +
+def comment_redaction(json_data):
+    """
+    Redact PII on comment lines on extracted json data.
+
+    """
+    data = json.loads(json_data)
+    json_docs = data.get("documents", [])
+    print("len of docs: ", len(json_docs))
+    print(json_docs)
+    for doc in json_docs:
+        print(doc)
+        if "text" in doc:
+            # print("initial", doc["text"])
+
+            # Extract the comment content between /* and */
+            start_idx = doc["text"].find("/*")
+            end_idx = doc["text"].find("*/")
+            if start_idx != -1 and end_idx != -1:
+                comment = doc["text"][start_idx : end_idx + 2]
+
+                # Create a Dask DataFrame from the given data
+                comment_df = dd.from_pandas(
+                    pd.DataFrame([{"text": comment}]), npartitions=1
+                )
+                # Create a DocumentDataset instance
+                comment_doc_dataset = DocumentDataset(comment_df)
```

Comment:
I'm a little confused as why this is being done this way. Couldn't we do this as part of the `run_curation_pipeline`? My concern is that this will have poor performance.

Reply:
It has poor performance. But also, redaction has a few bugs when used for code files. We discussed this with Arham as well. The idea is to redact only the comment lines and not all the code. We saw that proper names within code get redaxted/replaced otherwise. Is there an alternate way?

Reply:
Adding @ryantwolf and @ayushdg to comment on how to workaround this issue.

Reply:
As far as I know the PII tool was made to operate on normal text documents and not on code. In particular, it assumes it can operate on the entire document. So, there will probably need to be a workaround. However, I can suggest on how to improve performance. I would recommend you construct the entire `DocumentDataset` filled with every comment in your dataset. Then, perform the PII on the comments, then replace them. Doing this in a batched manner should allow the PII removal to perform better. @singhva if you have any additional thoughts

Reply:
Thanks @ryantwolf  and @Maghoumi . I have updated the PII reaction piece now.